src/storage/local.rs

```diff
@@ -0,0 +1,85 @@
+#![allow(missing_docs)]
+
+use std::fs;
+use std::vec::Vec;
+use std::path::{Path, PathBuf};
+
+use core::{LalResult,config_dir,ensure_dir_exists_fresh};
+
+
+#[derive(Serialize, Deserialize, Clone, Default)]
+pub struct LocalConfig {}
+
+use super::{Backend, Component};
+
+pub struct LocalBackend {
+    /// Local config
+    pub config: LocalConfig,
+    /// Cache directory
+    pub cache: String,
+}
+
+impl LocalBackend {
+    pub fn new(cfg: &LocalConfig, cache: &str) -> Self {
+        LocalBackend{
+            config: cfg.clone(),
+            cache: cache.into(),
+        }
+    }
+}
+
+/// Artifact backend trait for `LocalBackend`
+///
+/// This is intended to be used by the caching trait `CachedBackend`, but for
+/// specific low-level use cases, these methods can be used directly.
+impl Backend for LocalBackend {
+    fn get_versions(&self, _name: &str, _loc: Option<&str>) -> LalResult<Vec<u32>> {
+        Ok(vec![0])
+    }
+
+    fn get_latest_version(&self, _name: &str, _loc: Option<&str>) -> LalResult<u32> {
+        Ok(0)
+    }
+
+    fn get_tarball_url(
+        &self,
+        name: &str,
+        _version: Option<u32>,
+        _loc: Option<&str>,
+    ) -> LalResult<Component> {
+        let tar_path = format!("global/{}/0/{}.tar.gz", name, name);
+        let full_tar_path = config_dir().join(tar_path);
+
+        Ok(Component {
+                name: name.into(),
+                version: 0,
+                tarball: full_tar_path.into_os_string().into_string().unwrap()
```

Comment:
I'm not too happy with this unwrap/crash behaviour.
We should decide what to do if components have unicode names, at the very least they should be filesystem safe. Ideally, this function should fail with a suitable `LalResult`.

Reply:
i actually want to enforce lowercase ascii names as component names anyway, so you can really just use display and never think about this